### PR TITLE
Use library-go build scripts

### PR DESCRIPTION
**Pull request by sallyom**
This PR:
* makes switch to use library-go's build-scripts
* Makefile follows cluster-kube-apiserver-operator, with an added `make check` (verify,test-unit) to keep what's currently run in CI
* gets rid of cruft!
* updates CRD with nice descriptions! (bc verify won't pass w/out it) 
* bump(*) to bring in necessary library-go changes for `make images`
* adds ldflags to Dockerfile, Dockerfile.rhel7

### CODE REVIEW

**Comment by sallyom** [comment]
/hold

adding other fixes to make images

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,11 @@
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.12 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-authentication-operator
 COPY . .
-RUN go build -tags="ocp" -o authentication-operator ./cmd/authentication-operator
+ENV GO_PACKAGE github.com/openshift/cluster-authentication-operator
+RUN go build -ldflags "-X $GO_PACKAGE/pkg/version.versionFromGit=$(git describe --long --tags --abbrev=7 --match 'v[0-9]*')" -tags="ocp" -o authentication-operator ./cmd/authentication-operator
 
-FROM registry.svc.ci.openshift.org/openshift/origin-v4.2:base
```
Comment (sallyom) [inline Dockerfile]: v4.2, v4.1 do not exist, only v4.0 (other repos (KASop, OASop Dockerfile use the v4.0, also)

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
 ENTRYPOINT ["/usr/bin/authentication-operator"]
-LABEL io.k8s.display-name="OpenShift cluster-authentication-operator" \
-      io.k8s.description="This is a component of OpenShift and manages cluster authentication settings" \
-      com.redhat.component="cluster-authentication-operator" \
-      maintainer="OpenShift Auth Team <aos-auth-team@redhat.com>" \
-      name="openshift/ose-cluster-authentication-operator" \
-      version="v4.0.0" \
-      io.openshift.tags="openshift" \
-      io.openshift.release.operator=true
+LABEL io.openshift.release.operator=true
```
Comment (sallyom) [inline Dockerfile]: following suit w/ other repos labels

File: Dockerfile.rhel7 [hunk 3]
```diff
@@ -1,17 +1,11 @@
...
 ENTRYPOINT ["/usr/bin/authentication-operator"]
-LABEL io.k8s.display-name="OpenShift cluster-authentication-operator" \
-      io.k8s.description="This is a component of OpenShift and manages cluster authentication settings" \
-      com.redhat.component="cluster-authentication-operator" \
-      maintainer="OpenShift Auth Team <aos-auth-team@redhat.com>" \
-      name="openshift/ose-cluster-authentication-operator" \
-      version="v4.0.0" \
-      io.openshift.tags="openshift" \
-      io.openshift.release.operator=true
+LABEL io.openshift.release.operator=true
```
Comment (sallyom) [inline Dockerfile.rhel7]: with any changes to the file, need to make sure it will not disrupt/break anything with ART

File: Makefile [hunk 4]
```diff
@@ -30,10 +30,19 @@ clean:
...
+# Example:
+#   make test-e2e
 .PHONY: test-e2e
-test-e2e: GO_TEST_PACKAGES :=./test/e2e/...
-test-e2e: GO_TEST_FLAGS += -v
+# test-e2e: GO_TEST_PACKAGES :=./test/e2e/...
+# test-e2e: GO_TEST_FLAGS += -v
 test-e2e: test-unit
+test-e2e:
+	echo "needs e2e"
```
Comment (enj) [inline Makefile]: Please add a stub test instead.

Comment (sallyom) [reply Makefile]: done (I think it'll do,look ok?)

**Pull request merged**